sphinx/util/smartypants.py

```diff
@@ -9,21 +9,9 @@
     configurations and fixes. Replaces earlier smartypants version as used up
     to Sphinx 1.5.6.
 
-    :copyright: © 2010 Günter Milde,
-                original `SmartyPants`_: © 2003 John Gruber
-                smartypants.py:          © 2004, 2007 Chad Miller
-    :license: Released under the terms of the `2-Clause BSD license`_, in short:
-
-       Copying and distribution of this file, with or without modification,
-       are permitted in any medium without royalty provided the copyright
-       notices and this notice are preserved.
-       This file is offered as-is, without any warranty.
-
-    .. _SmartyPants: http://daringfireball.net/projects/smartypants/
-    .. _2-Clause BSD license: http://www.spdx.org/licenses/BSD-2-Clause
-
-    See the LICENSE file and the original docutils code for details.
-
+    :copyright: Copyright 2003-2010 by Günter Milde, John Gruber,
+                  Chad Miller.
```

Comment:
I'm not good at license and copyright. Is this modification allowed?

Reply:
Yup, I'm retaining the copyright and given that the license is the same as the main code, we don't need to explicitly call this out.

Reply:
Could you explain for clarification which "main code" you are referring too?
And how does this change relate to the general purpose of the PR?

Reply:
> Could you explain for clarification which "main code" you are referring too?

I'm referring to the bulk of the Sphinx code: this file is licensed as BSD 2-Clause, and Sphinx itself is licensed as BSD 2-Clause. It's not possible for the latter to change without getting permission from every contributor to Sphinx, which means it's never going to happen. We're safe here.

> And how does this change relate to the general purpose of the PR?

In [this commit](https://github.com/sphinx-doc/sphinx/pull/4105/commits/505e1483ce88d0a9579e3b16ae93b6c2541ef0be), I move a custom header check to a flake8 plugin. Previously we were able to skip this particular file from that check because it doesn't fit the standard format. However, we're not able to do this in flake8 without skipping the entire file or inserting a special check for this file in the plugin. Personally, I'd rather modify the header a small bit than do one of those two things.

Reply:
Thanks for explanation. For the header check, you mean flake8 compliance entailed removing all lines starting at https://github.com/sphinx-doc/sphinx/pull/4105/commits/505e1483ce88d0a9579e3b16ae93b6c2541ef0be#diff-148ee4b7705bb41cbeae2ac61d69e1d5L14 (about 12 lines)? there was no way to amend them in some less drastic way? Perhaps indeed they are redundant. Regarding copyright, the copyright notice in its original formulation is the one of the full Docutils `docutils/utils/smartquotes.py`. The Sphinx `smartypants.py` is a small portion of this, which is only included to backport from Docutils 0.14 some features to Sphinx users with earlier Docutils. I think most if not all of this code was contribued by Günter Milde. Thus I am not sure about the way the copyright notice is set-up here. But the previous paragraph explains the situation to some extent.

Reply:
> Thanks for explanation. For the header check, you mean flake8 compliance entailed removing all lines starting at 505e148#diff-148ee4b7705bb41cbeae2ac61d69e1d5L14 (about 12 lines)?

Correct :smile: 

> there was no way to amend them in some less drastic way?

Not that I could find. The check expects the license to be the last line of the docstring and the copyright to be the second last line thing before that, though this one can be multi-line. As mentioned above, I could special case this file but I really didn't any reason to do so.

> Regarding copyright, the copyright notice in its original formulation is the one of the full Docutils docutils/utils/smartquotes.py.

Yup, but this is the same license so I think we're good

> The Sphinx smartypants.py is a small portion of this, which is only included to backport from Docutils 0.14 some features to Sphinx users with earlier Docutils. I think most if not all of this code was contribued by Günter Milde. Thus I am not sure about the way the copyright notice is set-up here.

Yeah, I'm not sure about this. It's easier to just leave it the way it is, IMO

> But the previous paragraph explains the situation to some extent.

Reply:
well, I think the removal of the ``2-Clause BSD license`` short description is safe, as well as removal of the hyperlinks. Perhaps if copyright is allowed to be multi-line, indeed keeping it the way would be a bit safer than reformulating it.

Reply:
I meant keeping it the way it is, sorry, hit push key too fast.

Reply:
I should have been more precise: it's allowed to be multi-line, but those lines are only allowed to be comma-separated strings (see the `copyright_2_re` regex in `sphinx/checks.py` above). I _could_ soften this check, but it's there for a reason and I don't think we should really change it?

Reply:
I am a bit worried that flake8 normalization goes far here, but I am in no way knowledgeable about copyright, so I don't think this is blocking.

Reply:
arrggh...I meant to write: _I don't know enough to argue in anyway this could be blocking_.

Reply:
I've just gone and skipped the file. You're right - it is taking flake8 normalization a bit far

Reply:
I only worried about merging three copyright entries to one. It seems like they are team, and worked until 2010.

Reply:
I've reverted this change now and simply skipped the file. We can re-evaluate this later